### ISSUE TITLE
DatePicker: icon / trigger button not focusable by default

### ISSUE THREAD

**Issue by jepsar** [primefaces/primefaces#8953]
### Describe the bug

I don't like the behavior where the date picker is shown on focus. Instead I use:
```xhtml
<p:datePicker ...
              showIcon="true"                             
              showOnFocus="false"/>
```
This comes with the accessibility issue that the icon is not focusable.

### Reproducer

See above.

### Expected behavior

Icon should be focusable by adding `tabindex="0"`, or at least it should be a setting as with autocomplete. And the UI states should be applied to the icon button.

### PrimeFaces edition

Community

### PrimeFaces version

11.0.0

**Comment by melloware** [comment]
You can do this already with `buttonTabIndex`

```xml
            <p:datePicker value="#{testView.localDateTime}"
              showIcon="true"
              showOnFocus="false"
	      buttonTabindex="0"/>
```

However I did find that the widget disable/enable was not disabling the button along with the input.

**Comment by jepsar** [comment]
Hmm.. okay, I was not expecting that such an attribute existed, because the default is different compared to AutoComplete.

**Comment by melloware** [comment]
Yeah I think its so you can separate the Input tabIndex and button tab index and also choose some people don't want the button to be tabbable etc.  I remember we added this a while ago when someone else wanted it tabbable.

**Comment by melloware** [comment]
Looks like it originates from Calendar: https://github.com/primefaces/primefaces/issues/1771

**Comment by jepsar** [comment]
I think it should be focusable by default for better accessibility. Users that don't want that can use the attribute to disable it.

**Comment by jepsar** [comment]
Shall I tackle this?

**Comment by melloware** [comment]
sure!  I agree i think it should be accessible by default.

**Comment by melloware** [comment]
I think it might just be this line...

```js
var buttonIndex = this.cfg.buttonTabindex||this.jqEl.attr('tabindex');
```

to

```js
var buttonIndex = this.cfg.buttonTabindex||this.jqEl.attr('tabindex')||0;
if(buttonIndex >= 0) {
```

???
